lua/modules/api.lua

```diff
@@ -56,12 +56,31 @@ end
 
 -- getFriends ...
 -- callback(ok, friends, errMsg)
-mod.getFriends = function(_, callback)
+mod.getFriends = function(_, callback, fields, config)
 	if type(callback) ~= "function" then
-		error("api:getFriends(callback) - callback must be a function", 2)
+		error("api:getFriends(callback, fields, config) - callback must be a function", 2)
 		return
 	end
-	local req = HTTP:Get(mod.kApiAddr .. "/friend-relations", function(resp)
+
+	local defaultConfig = {
+		userID = "self",
+	}
+
+	config = require("config"):merge(defaultConfig, config)
+
+	local url = mod.kApiAddr .. "/users/" .. config.userID .. "/friends"
+
+	if type(fields) == "table" then
+		for i, field in ipairs(fields) do
+			if i == 1 then
+				url = url .. "?f=" .. field
+			else
+				url = url .. "&f=" .. field
+			end
+		end
```

Comment:
Shouldn't we have a function that does this? Takes a table and returns a string, I can see the same thing at line128 (without the `?`)

Reply:
Yes, we'll factorise it at some point. It's a bit out of scope for the PR.